test/test_imagej.py

```diff
@@ -107,6 +107,62 @@ def main(self):
         unittest.main()
 
 
+class TestXarrayConversion(unittest.TestCase):
+    def testCstyleArrayWithLabeledDimsConverts(self):
+        xarr = xr.DataArray(np.random.rand(5, 4, 3, 6, 12), dims=['T', 'Z', 'C', 'Y', 'X'],
+                             coords={'X': range(0, 12), 'Y': np.arange(0, 12, 2), 'C': ['R', 'G', 'B'],
+                                     'Z': np.arange(10, 50, 10), 'T': np.arange(0, 0.05, 0.01)},
+                             attrs={'Hello': 'Wrld'})
+
+        dataset = ij.py.to_java(xarr)
+        axes = [cast('net.imagej.axis.DefaultLinearAxis', dataset.axis(axnum)) for axnum in range(5)]
+        labels = [axis.type().getLabel() for axis in axes]
+        origins = [axis.origin() for axis in axes]
+        scales = [axis.scale() for axis in axes]
+
+        self.assertListEqual(origins, [0, 0, 0, 10, 0])
+        self.assertListEqual(scales, [1, 2, 1, 10, 0.01])
+
+        self.assertListEqual(list(reversed(xarr.dims)), labels)
```

Comment:
Should we make the xarray above use small case as per convention, and then use upper here? Or too much effort? We do test the translation implicitly in the other test.